addons/account/models/account_move_line.py

```diff
@@ -2214,6 +2214,7 @@ def _check_amls_exigibility_for_reconciliation(self, shadowed_aml_values=None):
                                     This is usefull if you want to preview the reconciliation before doing some changes
                                     on amls like changing a date or an account.
         """
+        self = self.filtered(lambda aml: not aml.matching_number.startswith("P") if aml.matching_number else not aml.reconcile)
```

Comment:
```suggestion
        partial_matching_numbers = self.mapped('matching_number').filtered(lambda mn: mn and mn.startswith("P")
        self = self.filtered(lambda aml: not aml.reconciled or not aml.matching_number in partial_matching_numbers)
```
Note: I don't like this shadowing of self